validator/slashing-protection/local/standard-protection-format/import.go

```diff
@@ -94,7 +94,13 @@ func validateMetadata(ctx context.Context, validatorDB db.Database, interchangeJ
 
 	// We need to verify the genesis validators root matches that of our chain data, otherwise
 	// the imported slashing protection JSON was created on a different chain.
-	// TODO(#7813): Add this check, very important!
+	gvr, err := rootFromHex(interchangeJSON.Metadata.GenesisValidatorsRoot)
+	if err != nil {
+		return fmt.Errorf("%#x is not a valid root: %v", interchangeJSON.Metadata.GenesisValidatorsRoot, err)
+	}
+	if err = validatorDB.SaveGenesisValidatorsRoot(ctx, gvr[:]); err != nil {
```

Comment:
This is incorrect, we should instead fetch the genesis validators root and compare it. We should not save what we get from the JSON file like this

Reply:
@rauljordan save detects difference and returns error. will change to fetch

Reply:
Yes but the name is misleading though, we should just do db.GenesisValidatorsRoot and do a simple comparison